### PR TITLE
[YUNIKORN-2423] Remove unnecessary boolean return value from the tracking code

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
UserTracker returns a boolean value about whether the increment/decrement was successful or not. This is no longer relevant and can be removed. The result is simplified code.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2423

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Review by chia7712 — COMMENTED** [review]
overall +1, some trivial comments left

File: pkg/scheduler/ugm/queue_tracker.go [hunk 1]
```diff
@@ -122,11 +120,9 @@ func (qt *QueueTracker) increaseTrackedResource(hierarchy []string, applicationI
 		zap.Bool("use wild card", qt.useWildCard),
 		zap.Stringer("total resource after increasing", qt.resourceUsage),
 		zap.Int("total applications after increasing", len(qt.runningApplications)))
-	return true
 }
 
-// Note: Lock free call. The Lock of the linked tracker (UserTracker and GroupTracker) should be held before calling this function.
```
[the remark this answers is not in the document]
Comment (pbacsko) [reply pkg/scheduler/ugm/queue_tracker.go:128]: Probably a random mistake. Will restore it.

File: pkg/scheduler/ugm/queue_tracker.go [hunk 2]
```diff
@@ -140,12 +136,9 @@ func (qt *QueueTracker) decreaseTrackedResource(hierarchy []string, applicationI
 		if qt.childQueueTrackers[childName] == nil {
 			log.Log(log.SchedUGM).Error("Child queueTracker tracker must be available in child queues map",
 				zap.String("child queueTracker name", childName))
-			return false, false
-		}
-		removeQT, decreased := qt.childQueueTrackers[childName].decreaseTrackedResource(hierarchy[1:], applicationID, usage, removeApp)
-		if !decreased {
-			return false, decreased
+			return false
```
Comment (pbacsko) [inline pkg/scheduler/ugm/queue_tracker.go:139]: Uncovered branch. Will file a JIRA to enhance UT coverage.

File: pkg/scheduler/ugm/group_tracker.go [hunk 3]
```diff
@@ -48,20 +48,20 @@ func newGroupTracker(groupName string, events *ugmEvents) *GroupTracker {
...
 	gt.applications[applicationID] = user
-	return gt.queueTracker.increaseTrackedResource(strings.Split(queuePath, configs.DOT), applicationID, group, usage)
+	gt.queueTracker.increaseTrackedResource(strings.Split(queuePath, configs.DOT), applicationID, group, usage)
 }
 
-func (gt *GroupTracker) decreaseTrackedResource(queuePath, applicationID string, usage *resources.Resource, removeApp bool) (bool, bool) {
+func (gt *GroupTracker) decreaseTrackedResource(queuePath, applicationID string, usage *resources.Resource, removeApp bool) bool {
 	if gt == nil {
-		return false, true
+		return false
```
Comment (pbacsko) [inline pkg/scheduler/ugm/group_tracker.go:64]: Uncovered branch, will file a JIRA to address coverage.

File: pkg/scheduler/ugm/manager.go [hunk 4]
```diff
@@ -113,14 +113,14 @@ func (m *Manager) DecreaseTrackedResource(queuePath, applicationID string, usage
...
-		return false
+		return
 	}
 
 	userTracker := m.GetUserTracker(user.User)
 	if userTracker == nil {
 		log.Log(log.SchedUGM).Error("user tracker must be available in userTrackers map",
 			zap.String("user", user.User))
-		return false
+		return
```
Comment (pbacsko) [inline pkg/scheduler/ugm/manager.go:123]: Will file a JIRA to address coverage.

File: pkg/scheduler/ugm/user_tracker.go [hunk 5]
```diff
@@ -56,35 +56,22 @@ func newUserTracker(userName string, ugmEvents *ugmEvents) *UserTracker {
...
-					zap.String("user", ut.userName))
-			}
-		}
-		return increasedGroupUsage
-	}
-	return increased
+	hierarchy := strings.Split(queuePath, configs.DOT)
+	ut.queueTracker.increaseTrackedResource(hierarchy, applicationID, user, usage)
+	gt := ut.appGroupTrackers[applicationID]
+	log.Log(log.SchedUGM).Debug("Increasing resource usage for group",
```
Comment (chia7712) [inline pkg/scheduler/ugm/user_tracker.go:66]: Maybe we should move this log info to `group_tracker#increaseTrackedResource` for consistency?

**Review by chia7712 — APPROVED** [review]
overall +1 and just one small question

File: pkg/scheduler/ugm/user_tracker.go (hunk 5 above)
Comment (pbacsko) [reply pkg/scheduler/ugm/user_tracker.go:66]: Logging and GroupTracker handling is a bit all over the place. Created https://issues.apache.org/jira/browse/YUNIKORN-2542 as a follow-up.

**Pull request closed without merging**